scripts/ldap-check.sh

```diff
@@ -54,14 +55,35 @@ UserFilter=`cat $config_file | jq -r .LdapSettings.UserFilter`
 EmailAttribute=`cat $config_file | jq -r .LdapSettings.EmailAttribute`
 UsernameAttribute=`cat $config_file | jq -r .LdapSettings.UsernameAttribute`
 IdAttribute=`cat $config_file | jq -r .LdapSettings.IdAttribute`
+GroupFilter=`cat $config_file | jq -r .LdapSettings.GroupFilter`
+GroupIdAttribute=`cat $config_file | jq -r .LdapSettings.GroupIdAttribute`
 
 if [[ -z ${UserFilter} ]]; then
-	UserFilter="($IdAttribute=$1)"
+	UserFilter="($IdAttribute=$2)"
 else
-	UserFilter="(&($IdAttribute=$1)$UserFilter)"
+	UserFilter="(&($IdAttribute=$2)$UserFilter)"
 fi
 
+if [[ -z ${GroupFilter} ]]; then
+	GroupFilter="($GroupIdAttribute=$2)"
```

Comment:
If there's no `GroupFilter` configured the default filter is used: `(|(objectClass=group)(objectClass=groupOfNames)(objectClass=groupOfUniqueNames))` [as found here in the Enterprise repo](https://github.com/mattermost/enterprise/blob/28662795da4f0dd5c932587686619b2a4d50e4fe/ldap/ldap_session.go#L17).

For example, for a case where `GroupIdAttribute` is set to `entryUUID`, the command `./scripts/ldap-check.sh -g b7e32a52-6cd3-1039-9cc9-0fa221def835` would need to use a  filter such as `(&(entryUUID=b7e32a52-6cd3-1039-9cc9-0fa221def835)(|(objectClass=group)(objectClass=groupOfNames)(objectClass=groupOfUniqueNames)))`.

Reply:
Oh okay, I didn't take the default case into consideration because I (maybe wrongfully) expected only people who have issues with their group filters to use this script, also because we don't ship it with the mattermost release archives and it only really comes up when support sends it to customers, but I could also be wrong about this and maybe it is being used by a lot more customers than I anticipated.

Do you think it would be good to add the default case to the script as well?

Reply:
@svelle Would you like to move forward with this as-is and then we can address the users without configured group filters at a later date, if we find it adds value? I don't want the value you have added here to be blocked.